source/extensions/common/workload_discovery/api.cc

```diff
@@ -46,10 +46,22 @@ Istio::Common::WorkloadMetadataObject convert(const istio::workload::Workload& w
   default:
     break;
   }
+  std::string ns = workload.namespace_();
+  std::string trust_domain = workload.trust_domain();
+  // Trust domain may be elided if it's equal to "cluster.local"
+  if (trust_domain.empty()) {
+    trust_domain = "cluster.local";
+  }
+  // The namespace may be elided if it's equal to "default"
+  if (ns.empty()) {
+    ns = "default";
```

Comment:
This sounds dangerous to imply default.

Reply:
The comments are straight from the WDS proto